trace_model/obgn-products/trace_obgn-product_graphsage_model.ipynb

```diff
@@ -0,0 +1,546 @@
+    "# will contain only edge_idx and size attributes\n",
+    "edge_list_0 = []\n",
+    "edge_list_1 = []\n",
+    "edge_list_2 = []\n",
+    "edge_adjs = []\n",
+    "for idx, e_idx in enumerate(dummy_adjs[0]):\n",
+    "    if idx == 0:\n",
+    "        edge_list_0.append(e_idx[0])\n",
+    "        #edge_list_0.append(e_idx[1])\n",
+    "        edge_list_0.append(torch.tensor(np.asarray(e_idx[2])))\n",
+    "    elif idx == 1:\n",
+    "        edge_list_1.append(e_idx[0])\n",
+    "        #edge_list_1.append(e_idx[1])\n",
+    "        edge_list_1.append(torch.tensor(np.asarray(e_idx[2])))\n",
+    "    else:\n",
+    "        edge_list_2.append(e_idx[0])\n",
+    "        #edge_list_2.append(e_idx[1])\n",
+    "        edge_list_2.append(torch.tensor(np.asarray(e_idx[2])))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "a6af5763",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "edge_index_0 = edge_list_0[0]\n",
+    "edge_size_0 = edge_list_0[1]\n",
+    "\n",
+    "edge_index_1 = edge_list_1[0]\n",
+    "edge_size_1 = edge_list_1[1]\n",
+    "\n",
+    "edge_index_2 = edge_list_2[0]\n",
+    "edge_size_2 = edge_list_2[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "8e7546ce",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "tensor([718, 153])"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "edge_size_0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "275fe5d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
+    "x = data.x.to(device)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "b09534f8",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "torch.Size([718, 100])"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# total number of nodes involved in the computation graph\n",
+    "x[dummy_n_ids[0]].shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "7dc1143d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "torch.Size([718, 100])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# lets create node dummy input for the trace\n",
+    "dummy_x = x[dummy_n_ids[0]]\n",
+    "print(dummy_x.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "0949bc0c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "torch.Size([1000, 100])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# padding nodes\n",
+    "max_nodes = 1000\n",
```

Comment:
To flexibly calculate the maximum possible neighborhood size for padding if we want to vary the neighborhood sizes, we may want to introduce a convenience function like below. For the current model (including self loops, so `n+1` per hop) the total size is `1056` - we may want to rerun the trace for this, wdyt?

```
import math

def neighborhood_sizes_to_max_nodes(sizes_list):
    return math.prod([s+1 for s in sizes_list])
```

Reply:
@geenen124 need to test this scenario, will let you know!!